### ISSUE TITLE
BUG - Not opening/acknowledging all files in a multi-file Gist

### ISSUE THREAD

**Issue by stvhwrd** [kenhowardpdx/vscode-gist#18]
**Operating System:**
```
                 -/+:.          
                :++++.          OS: 64bit Mac OS X 10.13.1 17B48
               /+++/.           Kernel: x86_64 Darwin 17.2.0
       .:-::- .+/:-``.::-       
    .:/++++++/::::/++++++/:`    
  .:///////////////////////:`   Shell: zsh 5.4.2
  ////////////////////////`     Resolution: 2880x1800
 -+++++++++++++++++++++++`      DE: Aqua
 /++++++++++++++++++++++/       WM: Quartz Compositor
 /sssssssssssssssssssssss.      
 :ssssssssssssssssssssssss-     CPU: Intel Core i7-4770HQ @ 2.20GHz
  osssssssssssssssssssssssso/`  GPU: Intel Iris Pro
  `syyyyyyyyyyyyyyyyyyyyyyyy+`  RAM: 9237MiB / 16384MiB
   `ossssssssssssssssssssss/
     :ooooooooooooooooooo+.
      `:+oo+/:-..-:/+o+/-
```

**Visual Studio Code:**
```
Code

Version 1.17.2 (1.17.2)

b813d12980308015bcd2b3a2f6efa5c810c33ba5

2017-10-16T13:43:44.213Z
```

**This is a**

- [ x] bug report

#### Description

When opening a gist with multiple files, only three files are acknowledged/displayed - the first two (gist files are arranged alphabetically) and the last one.  
See (8 MB!) GIF below for the result of opening a six-file gist.

![nov-02-2017 16-55-33](https://user-images.githubusercontent.com/4117011/32355947-366acc04-bfef-11e7-8b0b-ab7ede4c2fb3.gif)

It'd be great to have them all open as tabs in the same pane, rather than individually in side-by-side panes.

Thanks for an otherwise-awesome extension @kenhowardpdx!

**Comment by stvhwrd** [comment]
I'm looking for alternatives to [`splitEditor`](https://github.com/kenhowardpdx/vscode-gist/blob/93edeebc4da4442cc0fbc18060cff3b7b66b72a3/src/controllers/main.controller.ts#L91).

 Is there a better source of documentation for VSCode API/commands than the [keybindings page](https://code.visualstudio.com/docs/getstarted/keybindings)?

**Comment by kenhowardpdx** [comment]
Oh wow! Thanks for reporting. I can see in your GIF that "LEFT" cycles through the remaining files. Good find. I have been wondering if [Multi-root Workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces) would be a good fit for this extension. TBH I haven't looked into it much.

**Comment by eugene2candy** [comment]
I just installed this extension 10min ago. And have the exact same issue. 
My opinion to solve this: 
After `GIST: Open Block`,  we should have all the options in this specific block, which the same like the function `GIST: Insert Code Into Current File`. So we can choose which we want to open. For me, I don't need to open all the files at the same time... 
@kenhowardpdx

**Comment by kenhowardpdx** [comment]
@eugene2candy interesting suggestion. Maybe a prompt w/ list of files and an option for opening all?

I wonder if a better approach would be to leave "Open Block" as-is but add another command for "Open Block File".

Thoughts?

**Comment by stvhwrd** [comment]
Selecting individual files is a good idea.

There are times, though, that a user will want all files in a gist. In that case - what about downloading all files to a temporary directory, and using native directory handling in Code? Rather than attempting to open all files on import, you could just open the directory (named for the gist) stored in an OS temp file location.

**Comment by kenhowardpdx** [comment]
Let's keep this issue focused on the bug...

But feel free to open an issue for these feature requests. I'd be interested in the "Open Block File" feature, but better to discuss in its own thread.

**Comment by kenhowardpdx** [comment]
@stvhwrd @eugene2candy this is fixed in v1.1.5 https://github.com/kenhowardpdx/vscode-gist/releases/tag/v1.1.5
